### ISSUE TITLE
use it as a library

### ISSUE THREAD

**Issue by oilel** [dizda/fast-socks5#31]
Is there any way to use it as a library instead of building to binary?

**Comment by dizda** [comment]
yes sure, you can have a look at the examples/, they use it as a library.
eg. https://github.com/dizda/fast-socks5/blob/master/examples/simple_tcp_server.rs#L99

**Comment by oilel** [comment]
> yes sure, you can have a look at the examples/, they use it as a library. eg. https://github.com/dizda/fast-socks5/blob/master/examples/simple_tcp_server.rs#L99

How to make a socks proxy? Just like Onion Routing's socks proxy, Tor Browser connects to Tor's socks port. I don't know how to modify IP routing.

**Comment by yuguorui** [comment]
> > yes sure, you can have a look at the examples/, they use it as a library. eg. https://github.com/dizda/fast-socks5/blob/master/examples/simple_tcp_server.rs#L99
> 
> How to make a socks proxy? Just like Onion Routing's socks proxy, Tor Browser connects to Tor's socks port. I don't know how to modify IP routing.

You could refer to [this patch](https://github.com/yuguorui/fast-socks5/commit/146df26ccd6382b831c04870621456ba0a17d583) to make a socks5 proxy, I've used it to make own proxy.

If you feel these patch is useful, I would be happy to raise a PR to this origin repo.

**Comment by dizda** [comment]
Fixed in #59
